LibCore: Use poll() results for notifiers on Android too

The Android-only branch in EventLoopImplementationUnix::wait_for_events() no
longer compiles (post_event() takes an EventReceiver*), and it activated every
notifier after each wake-up regardless of poll()'s revents, which spins the
event loop. Bionic's poll() behaves like glibc's, so use the common path.

Requirement: ARCH-002

diff --git a/Libraries/LibCore/EventLoopImplementationUnix.cpp b/Libraries/LibCore/EventLoopImplementationUnix.cpp
index ca9c23ea..ccfd2b9e 100644
--- a/Libraries/LibCore/EventLoopImplementationUnix.cpp
+++ b/Libraries/LibCore/EventLoopImplementationUnix.cpp
@@ -314,10 +314,6 @@ try_select_again:
         for (size_t i = 1; i < thread_data.poll_fds.size(); ++i) {
             auto& notifier = *thread_data.notifiers[i];
 
-#ifdef AK_OS_ANDROID
-            // FIXME: Make the check work under Android, perhaps use ALooper.
-            ThreadEventQueue::current().post_event(notifier, Core::Event::Type::NotifierActivation);
-#else
             auto revents = thread_data.poll_fds[i].revents;
 
             NotificationType type = NotificationType::None;
@@ -334,7 +330,6 @@ try_select_again:
 
             if (type != NotificationType::None)
                 ThreadEventQueue::current().post_event(&notifier, Core::Event::Type::NotifierActivation);
-#endif
         }
     }
 
